source/tutorials/batch-inference/csv-batch-inference.rst

```diff
@@ -0,0 +1,160 @@
+
+    If you want to, you can train the required model by following the Keras example here: `<https://keras.io/examples/structured_data/structured_data_classification_from_scratch/>`_.
+
+Running on Valohai
+------------------
+
+To easily run our batch inference on Valohai, we will use it to run our code from the very beginning.
+
+If you don't already have a Valohai account, go to `<https://app.valohai.com/>`_ to create one for yourself.
+
+Create a new folder for our project, then run the following commands in the project folder:
+
+.. code-block:: bash
+
+    vh login
+    # fill in your username
+    # and your password
+
+    vh init
+    # Answer the wizard questions like this:
+    # "First, let's..." -> y
+    # "Looks like..." -> python batch_inference.py, then y to confirm
+    # "Choose a number or..." -> tensorflow/tensorflow:2.5.0, then y to confirm
+    # "Write this to..." -> y
+    # "Do you want to link..." -> C, then give a name for your project, then select your user
+
+Edit the generated ``valohai.yaml`` so that it looks like this:
+
+.. code-block:: yaml
+
+    ---
+
+    - step:
+        name: Batch Inference
+        image: tensorflow/tensorflow:2.5.0
+        command:
+        - pip install pandas valohai-utils
+        - python batch_inference.py
+        inputs:
+        - name: model
+          default: s3://valohai-public-files/tutorials/batch-inference/csv-batch-inference/model.zip
+        - name: images
+          default: s3://valohai-public-files/tutorials/batch-inference/csv-batch-inference/data.csv
+
+What we are doing here is defining a single step for our machine learning pipeline, which is the Batch Inference step. We will run on top of the official ``tensorflow/tensorflow:2.5.0`` Docker image, first install the ``valohai-utils`` Python library and then run our batch inference code.
+
+Let's test that everything is set up correctly by running on Valohai:
+
+.. code-block:: bash
+
+    vh exec run --adhoc "Batch Inference"
+
+If everything went as planned, we should see our Valohai execution end after finding out that ``batch_inference.py`` is missing:
+
+.. image:: batch-inference-tutorial-1.png
+   :alt: Error, but success!
+
+Unpacking the Model
+--------------------
+
+Today we are unpacking the model ourselves. Let's get started by creating and opening up ``batch_inference.py`` in your favorite editor!
+
+Add these imports to the beginning of the file:
+
+.. code-block:: python
+
+    import json
+    from zipfile import ZipFile
+
+    import pandas as pd
+    import tensorflow as tf
+    import valohai as vh
+
+For unpacking the model, we will only need `zipfile` and `valohai`, but we will use the rest of the imports soon enough.
+
+Next, unpack the model to a folder called model in the current working directory:
+
+.. code-block:: python
+
+    with ZipFile(vh.inputs('model').path(process_archives=False), 'r') as f:
+        f.extractall('model')
```

Comment:
This will actually be automatically solved once https://github.com/valohai/valohai-utils/issues/45 is done.

I took it under progress now, but we want the CSV tutorial out immediately, which is why I did it manually here. Will update the tutorial later.